### PR TITLE
Make disabilities code less dumb

### PR DESCRIPTION
**Pull request by adacovsk**
[system][bugfix]

- Currently, there's `prob(x)` followed by `switch(rand(a,b)` convoluted nonsense

## What this does
- Cleans up the code a little bit, makes it more readable, less operations, and less convoluted
- Doesn't add/remove any phrases, and all probabilties should still be consistent

## Why it's good
Refactor bad code

### CODE REVIEW

**Comment by Eneocho** [comment]
>Pay homage to a realest thugs and fallen bros

a moment of silence for the fallen kings

**Comment by BomberBro** [comment]
I can't comment on the code changes, but the brain damage additions of:
"I CANT BREET", "NIGGER", and "TROON" feel pretty pointless (and are 75% of the additions). While one is just a modern reference to political events, which is dumb and out of place, the other two are just kind of buzzwordy bad words for the sake of bad words. I don't want to constantly hear "NIGGER TROON NIGGER" if somebody gets tourettes like I'm in a particularly bad /v/ thread. 

Frankly, just because we're a server where you can say funny words doesn't mean code has to be modeled after the Raimi cut. 
Edit: Also something something unatomic PR considering it handles a rework THEN adds lines to the brain damage list.

**Comment by adacovsk** [comment]
> Frankly, just because we're a server where you can say funny words doesn't mean code has to be modeled after the Raimi cut.

removed the tourettes changes

**Comment by Eneocho** [comment]
You should've stood by your changes and not bent the knee. It is too late now.

**Comment by N3X15** [comment]
```dm
if(prob(x))
  // a
if(prob(y))
  // b
if(prob(z))
  // c
```
means there's a probability `b` and `c` can run even if `a` is invoked.  If this is intended, fine, otherwise you may want to replace the later `if`s with `else if`

Also, we should consider moving braindamage shit to a textfile or something.

**Comment by jwhitak** [comment]
Violates the unwritten "one year" meme rule (IT'S MORBIN TIME! / I can't breet).

Edit: I can't breet is basically /vg/ server culture at this point so I'll vouch for that one. We're gonna forget poor Morbius in a year though. Sorry.

**Comment by adacovsk** [comment]
no more morb

**Comment by adacovsk** [comment]
> textfile or something.

Are there examples of where this is implemented?

**Comment by Hinaichigo** [comment]
From what I understand, switch is also nicer than if/else because it's a little more efficent. I don't really see how moving to if/else is an improvement.

**Comment by adacovsk** [comment]
> From what I understand, switch is also nicer than if/else because it's a little more efficent. I don't really see how moving to if/else is an improvement.

this removes additional operations
`
		if(prob(3))
			switch(pick(1,2,3))
				if(1)`
 to `if(prob(3))`
also, I'm replacing `		var/rn = rand(0, 200) //This is fucking retarded, but I'm only doing sanitization, I don't have three months to spare to fix everything
		if(getBrainLoss() >= 5)
			if(0 <= rn && rn <= 3)`

with `getBrainLoss > whatever && prob(1.5)` ... This is better in every way

**Comment by Eneocho** [comment]
Lame. Return the breet line. It's server culture.

**Comment by jwhitak** [comment]
Atomic changes. Return the breet line in another PR. I will updoot and recommend speed merge.

File: code/modules/mob/living/carbon/human/life/handle_disabilities.dm [hunk 1]
```diff
@@ -1,137 +1,135 @@
...
-		var/affect_chance = 100
-		var/affect_amount = 0
+	if(stat == DEAD)
+		return
+	if(disabilities & ELECTROSENSE && !dna.GetSEState(JAMSIGNALSBLOCK))
+		var/affect_chance = 30
+		var/affected = FALSE
 		if(head && istype(head,/obj/item/clothing/head/tinfoil))
-			affect_chance /= 2
+			affect_chance -= 15
```
Comment (SECBATON-GRIFFON) [inline code/modules/mob/living/carbon/human/life/handle_disabilities.dm]: i'd rather you kept the divides, so the final result is 7.5 and not 1, which seems way too low. the lowering to thirty in base is okay

Comment (adacovsk) [reply code/modules/mob/living/carbon/human/life/handle_disabilities.dm]: 1% is still pretty high, consider asthma attacks are 0.2%... 7.5% is greater than any other disability already, so the protection is basically useless

Comment (adacovsk) [reply code/modules/mob/living/carbon/human/life/handle_disabilities.dm]: the halloss is already debilitating enough that rye is necessary, whereas you can ignore most other disabilities

**Comment by Hinaichigo** [comment]
Why remove the chance for `;` to say it over the radio?

**Comment by adacovsk** [comment]
> Why remove the chance for `;` to say it over the radio?

It didn't work, I'll try to add it in the next PR

**Comment by jwhitak** [comment]
Cursed PR with too many changes at this point. The emojis are useless because this PR has changed what it changes more than 3 times. It was a code refactor and some brain damage/disability changes. Then it was just a code refactor. Now it's a code refactor and a disability 'bugfix'. Unatomic to atomic and then back to unatomic.

I want to underline this because this is sneakycoding. You are doing a relatively major(?) and arguably unrelated change to a PR that a lot of folks have already reviewed and made a decision on. In this case, it'd probably get more positive emoji since the rare paincrit disability sucks, but if folks didn't want this bugfix, it would be slipping it unfairly into the code without fair oversight.

When you make a PR, it's supposed to be one set of changes that people come and review. Editing and changing the code is meant to fix oversights/bugs in that set of changes, not continuously change the PR's list of changes. I'd say this PR should die because it's been changed up too many times in what it does. Kill this off and make a new one.

Ideally, make three new PRs. One for the code refactor, one for the paincrit bugfix, and one for the I CAN'T BREET brain damage messaging. All of which will have new emoji. I know that a lot of us are ragging on you for this, but I don't want to have to keep opening your PRs every day to see what has dramatically changed since the last time I've seen them.

**Comment by adacovsk** [comment]
okay, I'll separate it

**Comment by jwhitak** [comment]
Ok, we're back to a nice, atomic, code refactor. Updoot restored.

**Comment by Kurfursten** [comment]
confl

**Comment by Kurfursten** [comment]
jwhit's comment is very true. Also, as a number of my colleagues have recently pointed out, relying too much on emojis is dangerous - not in the least because of jwhit brings up (PRs can change). Having looked this over, I feel satisfied it is now atomic and that if people were to cast votes again, it would not be controversial. If someone vehemently disagrees with me, feel free to PR.

**Pull request merged**